core/templates/components/state-editor/state-responses-editor/state-responses.component.spec.ts

```diff
@@ -0,0 +1,994 @@
+    expect($scope.activeAnswerGroupIndex).toBe(undefined);
+    expect($scope.inapplicableSkillMisconceptionIds).toEqual(undefined);
+
+    ctrl.$onInit();
+
+    expect($scope.inapplicableSkillMisconceptionIds)
+      .toEqual(['misconception1']);
+
+    onAnswerGroupsChangedEmitter.emit();
+
+    expect($scope.answerGroups).toEqual(answerGroups);
+    expect($scope.defaultOutcome).toEqual(defaultOutcome);
+    expect($scope.activeAnswerGroupIndex).toBe(0);
+    expect($scope.inapplicableSkillMisconceptionIds).toEqual([]);
+
+    ctrl.$onDestroy();
+  });
+
+  it('should update answer choices', () => {
+    let onUpdateAnswerChoicesEmitter = new EventEmitter();
+    spyOnProperty(StateEditorService, 'onUpdateAnswerChoices')
+      .and.returnValue(onUpdateAnswerChoicesEmitter);
+    spyOn(ResponsesService, 'updateAnswerChoices');
+
+    ctrl.$onInit();
+    onUpdateAnswerChoicesEmitter.emit();
+
+    expect(ResponsesService.updateAnswerChoices).toHaveBeenCalled();
+
+    ctrl.$onDestroy();
+  });
+
+  it('should update custom arguments', () => {
+    let onHandleCustomArgsUpdateEmitter = new EventEmitter();
+    spyOnProperty(StateEditorService, 'onHandleCustomArgsUpdate')
+      .and.returnValue(onHandleCustomArgsUpdateEmitter);
+    spyOn(ResponsesService, 'handleCustomArgsUpdate').and.callFake(
+      (newAnswerChoices, callback) => {
+        callback();
+      }
+    );
+
+    ctrl.$onInit();
+    onHandleCustomArgsUpdateEmitter.emit();
+
+    expect(ctrl.onSaveInteractionAnswerGroups).toHaveBeenCalled();
+
+    ctrl.$onDestroy();
+  });
+
+  it('should set misconceptions when state editor is initialized', () => {
+    let onStateEditorInitializedEmitter = new EventEmitter();
+    spyOnProperty(StateEditorService, 'onStateEditorInitialized')
+      .and.returnValue(onStateEditorInitializedEmitter);
+    spyOn(StateEditorService, 'getMisconceptionsBySkill')
+      .and.returnValue({
+        skill1: [misconceptionObjectFactory.create(
+          'm1', 'Misconception 1', 'note', '', false)]
+      });
+
+    expect($scope.misconceptionsBySkill).toBe(undefined);
+    expect($scope.containsOptionalMisconceptions).toBe(undefined);
+
+    ctrl.$onInit();
+    onStateEditorInitializedEmitter.emit();
+
+    expect($scope.misconceptionsBySkill).toEqual({
+      skill1: [misconceptionObjectFactory.create(
+        'm1', 'Misconception 1', 'note', '', false)]
+    });
+    expect($scope.containsOptionalMisconceptions).toBe(true);
+
+    ctrl.$onDestroy();
+  });
+
+  it('should set active answer group index as -1 when starting sorting', () => {
+    let ui = {
+      placeholder: {
+        height: () => {}
+      },
+      item: {
+        height: () => {}
+      }
+    };
+
+    spyOn(ResponsesService, 'getActiveAnswerGroupIndex').and.returnValue(-1);
+    spyOn(ExternalSaveService.onExternalSave, 'emit');
+
+    ctrl.$onInit();
+    $scope.ANSWER_GROUP_LIST_SORTABLE_OPTIONS.start('', ui);
+
+    expect($scope.activeAnswerGroupIndex).toBe(-1);
```

Comment:
Pre-check for `$scope.activeAnswerGroupIndex`.

Reply:
Done